doc/design/zarr-manifests.md

```diff
@@ -0,0 +1,200 @@
+
+    - The `entries` object itself represents the top level directory of the
+      Zarr.
+
+    For example, a Zarr with the following structure:
+
+    ```text
+    .
+    ├── .zgroup
+    ├── arr_0/
+    │   ├── .zarray
+    │   └── 0
+    └── arr_1/
+        ├── .zarray
+        └── 0
+    ```
+
+    would have an `entries` field as follows (with elements of the entry arrays
+    omitted):
+
+    ```json
+    {
+        ".zgroup": [ ... ],
+        "arr_0": {
+            ".zarray": [ ... ],
+            "0": [ ... ]
+        },
+        "arr_1": {
+            ".zarray": [ ... ],
+            "0": [ ... ]
+        }
+    }
+    ```
+
+> [!NOTE]
+> The manifest files created by @yarikoptic contain the following fields which
+> are not present in the format described above:
+>
+> - A top-level `schemaVersion` key with a constant value of `2`
+>
+> - A `zarrChecksumMismatch` field inside the `statistics` object, used to
+>   store the checksum that the API reports for a Zarr when it disagrees with
+>   the checksum calculated by the manifest-generation code
+
+
+Archive API Changes
+-------------------
+
+***WIP***
+
+* Zarr version IDs equal the Zarr checksum
+
+* Asset properties gain `zarr_version: str | null` field (absent or null if Zarr is not yet ingested or asset is not a Zarr)
+    - Not settable by client
+    - Mint new asset when version changes?
+
+* Add `zarr_version` field to …/assets/path/ results
+
+* Zarr `contentUrl`s:
+    - Make API download URLs for Zarrs redirect to dandidav
+    - Replace S3 URLs with webdav.{archive_domain}/zarr/ URLs?
+        - Document needed changes to dandidav?
+            - The bucket for the Archive instance will now be given on the command line (only required if a custom/non-default API URL is given)
+            - The bucket's region will have to be looked up & stored before starting the webserver
+            - Zarrs under `/dandisets/` will no longer determine their S3 location via `contentUrl`; instead, they will combine the Archive's bucket & region with the Zarr ID in the asset properties (templated into "zarr/{zarr_id}/")
+
+* Getting specific Zarr versions & their files from API endpoints
+    - The current `/zarr/{zarr_id}/…` endpoints operate on the most recent version of the Zarr
+    - `GET /zarr/{zarr_id}/versions/` (paginated)
+    - `GET /zarr/{zarr_id}/versions/{version_id}/` ?
+    - `GET /zarr/{zarr_id}/versions/{version_id}/files/[?prefix=...]` (paginated)
+        - The Zarr entry objects returned in `…/files/` responses (with & without `versions/{version_id}/`) will need to gain a `VersionId` field containing the S3 object version ID
+    - Nothing under /zarr/versions/ is writable over the API
+
+* Publishing Zarrs: Just ensure that the `zarr_version` in Zarr assets is frozen and that no entries/S3 object versions from the referenced version are ever deleted ?
+
+* Does garbage collection of old Zarr versions need to be discussed?
```

Comment:
I think it should indeed be touched on. And we should touch on ["trailing delete"](https://github.com/dandi/dandi-archive/blob/0a3524d774baf2d0df4c820fb0509af7b68bafcc/doc/design/s3-trailing-delete.md?plain=1#L57) to be disabled for `/zarr/` prefix on s3.